### ISSUE TITLE
Question about the Performance of CCT6/3x2

### ISSUE THREAD

**Issue by DacDinh147** [SHI-Labs/Compact-Transformers#24]
Firstly, Thank for your great work. 
I trained your CCT6 on google colab and the result is
```
[Epoch 200][Train][100] 	 Loss: 5.9658e-01 	 Top-1  95.77
[Epoch 200][Train][110] 	 Loss: 5.9784e-01 	 Top-1  95.70
[Epoch 200][Train][120] 	 Loss: 5.9848e-01 	 Top-1  95.67
[Epoch 200][Train][130] 	 Loss: 5.9910e-01 	 Top-1  95.61
[Epoch 200][Train][140] 	 Loss: 6.0073e-01 	 Top-1  95.55
[Epoch 200][Train][150] 	 Loss: 6.0226e-01 	 Top-1  95.46
[Epoch 200][Train][160] 	 Loss: 6.0283e-01 	 Top-1  95.46
[Epoch 200][Train][170] 	 Loss: 6.0209e-01 	 Top-1  95.50
[Epoch 200][Train][180] 	 Loss: 6.0230e-01 	 Top-1  95.50
[Epoch 200][Train][190] 	 Loss: 6.0174e-01 	 Top-1  95.52
[Epoch 200][Train][200] 	 Loss: 6.0133e-01 	 Top-1  95.53
[Epoch 200][Train][210] 	 Loss: 6.0177e-01 	 Top-1  95.50
[Epoch 200][Train][220] 	 Loss: 6.0121e-01 	 Top-1  95.51
[Epoch 200][Train][230] 	 Loss: 6.0156e-01 	 Top-1  95.49
[Epoch 200][Train][240] 	 Loss: 6.0092e-01 	 Top-1  95.52
[Epoch 200][Train][250] 	 Loss: 6.0064e-01 	 Top-1  95.54
[Epoch 200][Train][260] 	 Loss: 6.0068e-01 	 Top-1  95.54
[Epoch 200][Train][270] 	 Loss: 6.0083e-01 	 Top-1  95.52
[Epoch 200][Train][280] 	 Loss: 6.0039e-01 	 Top-1  95.54
[Epoch 200][Train][290] 	 Loss: 6.0092e-01 	 Top-1  95.52
[Epoch 200][Train][300] 	 Loss: 6.0109e-01 	 Top-1  95.52
[Epoch 200][Train][310] 	 Loss: 6.0116e-01 	 Top-1  95.53
[Epoch 200][Train][320] 	 Loss: 6.0130e-01 	 Top-1  95.52
[Epoch 200][Train][330] 	 Loss: 6.0153e-01 	 Top-1  95.51
[Epoch 200][Train][340] 	 Loss: 6.0245e-01 	 Top-1  95.46
[Epoch 200][Train][350] 	 Loss: 6.0211e-01 	 Top-1  95.47
[Epoch 200][Train][360] 	 Loss: 6.0238e-01 	 Top-1  95.47
[Epoch 200][Train][370] 	 Loss: 6.0208e-01 	 Top-1  95.49
[Epoch 200][Train][380] 	 Loss: 6.0174e-01 	 Top-1  95.51
[Epoch 200][Train][390] 	 Loss: 6.0167e-01 	 Top-1  95.51
[Epoch 200][Eval][0] 	 Loss: 6.7492e-01 	 Top-1  93.75
[Epoch 200][Eval][10] 	 Loss: 6.7268e-01 	 Top-1  93.39
[Epoch 200][Eval][20] 	 Loss: 6.7867e-01 	 Top-1  93.42
[Epoch 200][Eval][30] 	 Loss: 6.7773e-01 	 Top-1  93.42
[Epoch 200][Eval][40] 	 Loss: 6.8292e-01 	 Top-1  93.43
[Epoch 200][Eval][50] 	 Loss: 6.7983e-01 	 Top-1  93.41
[Epoch 200][Eval][60] 	 Loss: 6.8163e-01 	 Top-1  93.37
[Epoch 200][Eval][70] 	 Loss: 6.7865e-01 	 Top-1  93.43
[Epoch 200] 	 	 Top-1  93.44 	 	 Time: 149.51
Script finished in 149.51 minutes, best top-1: 93.48, final top-1: 93.44
```

The accuracy on training set is always higher than on validation set. The model is saturated at about 91% accuracy on validation set. Is it acceptable? or the model losses its generalization or something overfitting.

**Comment by alihassanijr** [comment]
Hi,
Thank you for your interest.
I'm confused, it looks like the model is reaching over 93%, which is similar to what we reported.
What seems to be the issue?

**Comment by DacDinh147** [comment]
Hi, I means the accuracy of on the training set always higher than the accuracy of the validation set. It performs up to 95.xx% on training set but only get 93.xx% on the validation set. Is it a sign of loss generalization?.

**Comment by stevenwalton** [comment]
The training set is almost always going to have higher accuracy than the validation set in any statistical model you use (deep learning or not). This is normal. This is because a model is learning on a fixed set of data but being tested on things it hasn't seen before, which are going to have some slight deviations from the train set, even though they should exist within the same probability distribution. To determine overfitting we actually need to see if we continue learning on the training set but not the validation set. During this time the accuracy on the validation set will actually start decreasing. That will indicate that you have over trained your model (this is again general to any statistical learning method and not just neural networks).

**Comment by DacDinh147** [comment]
Thank for your explaination @stevenwalton. I close the question.
